tests/testthat/test-data.table.R

```diff
@@ -69,6 +75,51 @@ test_that("functions that use data.table do not modify input data table", {
   cut_data_by_date(x, 5)
   expect_identical(x, x_original)
 
+  # early_zero_weight()
+  x <- sim_pw_surv(n = 200)
+  x <- cut_data_by_event(x, 125)
+  x <- counting_process(x, arm = "experimental")
+  data.table::setDT(x)
+  x_original <- data.table::copy(x)
+  early_zero_weight(x, early_period = 2)
+  expect_identical(x, x_original)
+  # stratified
+  x <- data.frame(
+    stratum = rep(c("Biomarker-negative", "Biomarker-positive"), c(3L, 6L)),
+    events = rep(1, 9L),
+    n_event_tol = c(0, 1, 1, 0, 1, 0, 1, 1, 0),
+    tte = c(
+      1.5993802300052806, 2.0909139412967335, 10.506616675248615,
```

Comment:
After a while, people would wonder where all these magic numbers came from - I'd put the code to generate them here for clarity.

Reply:
The code to produce the input data is 40+ lines of code. I put the full code to produce it in the `early_zero_weight()` tests. However, these data.table tests are simply to confirm the data.table behavior is working correctly, so I try to minimize the amount of extraneous setup code.

Reply:
How about I add a comment stating that this data was created using the code that is available from the example in the function documentation?

Reply:
Actually I don't care that strongly about this. I went ahead and duplicated the code